Format All Facts Page

UtkarshKr007 opened the pull request:
The all facts page can be toggled between grid and list view. Right now each are in their own functions.  Next step is moving the into a component.

Commit:
Create all facts page with grid

Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

Commit:
Add link to facts.js that links to grid-facts.js

Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

Commit:
Install packages for "@material-ui/lab" 

Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

Commit:
Add toggle Button for fact.js and grid-fact.js

Added a toggle Button to toggle between the all facts pages. We are considering combining the pages into one and using css to hide one of the pages html when that respective button is clicked.  Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

UtkarshKr007 committed:
Add toggle between grid or list for all facts page

Co-authored-by: Shawn Reuter <reute078@morris.umn.edu>

UtkarshKr007 committed:
Make grid view a component

UtkarshKr007 committed:
Make List View a component

UtkarshKr007 committed:
Replace the old facts page with the new layout

I deleted the old facts.js page and refactored the grid-facts.js
filename,query and functions so it's regarded as the new facts page

UtkarshKr007 commented:
The GridView and ListView are now each in a component of their own rather be function on fact page which increases code readability. Next step would be to decide the location and styling of the toggle buttons. After that individual styling of the components themselves if needed.

UtkarshKr007 committed:
Replace view buttons with single boolean button

Now users can switch between layout and grid view by clicking a single
button rather than clicking on the button for that layout.

UtkarshKr007 committed:
Disable toggling on mobile. Make list view default

UtkarshKr007 committed:
Add excerpt to fact list

UtkarshKr007 committed:
Remove category from small fact card. Add tooltip.

Aditionally, change toggle button to icon button and align title

floogulinc commented:
I wonder if its possible to remember the grid vs list view setting in memory so that when you go to a fact from grid view and then back to the list it doesn't reset the view.

UtkarshKr007 committed:
Move hook inside React Context & wrap around app

I moved the hook that was determining the grid & list view out of facts
page and put it inside a react context in view-provider.js. The context
provider providers the state & function to change state. The context
consumer allows facts page to use the state. Gatsby-browser allows us to
use wrapRootElement to wrap the app inside the context so every page
has the context which includes the state controlling the view.

Commit:
Merge branch 'master' into facts-page

UtkarshKr007 committed:
Make media query styling conditional based on view

UtkarshKr007 committed:
Make ToggleLayout component using view components

ToggleLayout component contains the page title, button for toggling
views and both grid view and list view components. The different views
with toggling can easily be implemented on any page using this component

UtkarshKr007 commented:
Forgot to co-author, but worked with @reuterShawn on making the ToggleLayout component which contains the smaller view components as well as the button controlling the view in the commit above.

Commit:
Change layout to be wider for grid-view

Changed the maxWidth to be large instead of medium to allow for more cards to fit on one line. Also changed margin of toggle-layout component to be 8px

Commit:
Change layout to maxWidth large

Changed container to maxWidth large to allow for more facts to fit on one line in grid view also changed margin to 8px.

Commit:
Merge remote-tracking branch 'origin/facts-page' into facts-page

Commit:
Change margin  to 16px

reuterShawn commented:
> You can revert this change.

Sorry Paul I originally undid my commit but then when I did the right changes and commited them Gitkraken asked me to pull the branch up and then it just pulled back my commit again even though I discarded the changes on my commit. Anyways I put the other margin back to 16px on another commit what a waste of build minutes.

floogulinc reviewed (approved):
Looks great!

floogulinc commented on src/pages/facts.js:
```diff
@@ -2,81 +2,50 @@ import React from "react"
+                listImage: fixed(width: 40, height: 40) {
+                  ...GatsbyImageSharpFixed_withWebp
+                }
+                fixed(width: 190, height:150) {
```
```suggestion
                fixed(width: 190, height: 150, quality: 70) {
```

floogulinc commented on src/pages/facts.js:
```diff
@@ -2,81 +2,50 @@ import React from "react"
+          image {
+            src {
+              childImageSharp {
+                listImage: fixed(width: 40, height: 40) {
```
```suggestion
                listImage: fixed(width: 40, height: 40, quality: 70) {
```

kklamberty reviewed (approved):
Looks great! Are you going to add those suggestions from @floogulinc or no? I don't know what that would do, but I do like the overall look of these views. Nice work!

UtkarshKr007 committed:
Update src/pages/facts.js

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

UtkarshKr007 committed:
Update src/pages/facts.js

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>
